jina_commons/logging.py

```diff
@@ -0,0 +1,59 @@
+import os
+import time
+from typing import List
+
+from jina import DocumentArray
+from jina.logging.logger import JinaLogger
+
+
+def _get_non_empty_fields_doc_array(docs: DocumentArray) -> List[str]:
+    non_empty_fields = list(docs[0].non_empty_fields)
+    for doc in docs[:1]:
+        for field in non_empty_fields:
+            if field not in doc.non_empty_fields:
+                non_empty_fields.pop(field)
+    return non_empty_fields
+
+
+def add_request_logger(logger):
+    """
+    Add logging functionality to a request function.
+    Only shows logs for `JINA_LOG_LEVEL` > info.
+    You can set this as an env variable before starting your `Jina` application.
+
+    Example usages:
+    >>> from jina import Executor, requests
+    >>> my_logger = JinaLogger('MyExecLogger')
+    >>>
+    >>> class MyExec(Executor):
+    >>>     @requests
+    >>>     @add_request_logger(my_logger)
+    >>>     def index(self, docs, parameters, **kwargs):
+    >>>          ...
+
+    :param logger: The logger you want to use
+    """
+    def decorator(function):
+        def wrapper(self, docs, parameters, **kwargs):
+
+            if os.environ.get('JINA_LOG_LEVEL') in ['info', None]:
+                return function(self, docs, parameters, **kwargs)
```

Comment:
Note that there are other log levels that have higher rank then `INFO`. https://github.com/jina-ai/jina/blob/master/jina/enums.py#L136
```suggestion
            from jina.enums import LogVerbosity
            verbose_level = os.environ.get('JINA_LOG_LEVEL', None)
            verbose_level = LogVerbosity.from_string(verbose_level) if verbose_level is not None
            if verbose_level is None or verbose_level < LogVerbosity.INFO:
                return function(self, docs, parameters, **kwags)
```

Reply:
That is true, fixed in https://github.com/jina-ai/jina-commons/pull/20/commits/ddf953419a913316ea188b6844b96440ef86a8d5